phplib/Constant.php

```diff
@@ -35,12 +35,14 @@ class Constant {
     // Replace the ordinal indicator with the degree sign.
     '/(?<!\\\\)º/'   =>  '°',    /* U+00BA => U+00B0 */
 
+    // romanian quoting style
+    '/(?<!\\\\)"([^"]*)"/' => '„$1”',/* "x" => „x” */
```

Comment:
Aș prefera să nu facem asta la cleanup. Ideea e că, dacă salvezi o dată greșit (să zicem că uiți prima ghilimea), ți le pune pe toate celelalte aiurea: abc„ ”def„ ”ghi„ etc.

Reply:
Nu știu unde mi-a fost capul. Nici nu are ce căuta acolo.

Reply:
Trebuie rezolvat următorul clinci:
 Dacă htmlizez abrevierile prima dată, ca să pot face
`preg_match_all("/(?<!\\\\)#([^#]*)#/", $s, $matches, PREG_OFFSET_CAPTURE);`
din `htmlizeAbbreviations` îmi vine pregătit șirul cu `... class="abbrev" data-html="true"...` pe care bucla cu `HTML_PATTERNS` mi-o face praf.
Dacă las bucla șabloanelor html înaintea htmlizării abrevierilor nu mai găsesc ceea ce caut.
Propui tu ceva? că ai mai multă experiență.

Reply:
Ca să nu mai amintesc faptul că și notele de subsol ar putea avea abrevieri.

Reply:
Este OK să aibă abrevieri - fiecare notă de subsol extrasă dintre `{{...}}` este la rândul ei trecută prin `htmlize()`.

Reply:
În **dlrlv-ext.conf** avem _A SEC. XVIII^{1}_ ca abreviere. De aceea am mutat `Abbrev::htmlizeAbbreviations` mai sus, să nu trebuiască să reconvertesc în format intern definiția și să compar. Altfel trebuie abrevierea s-o htmlizez și astfel să fac `preg_match`-ul corespunzător.

Reply:
Nu știu de ce nu pot comenta la „De ce nu mai găsești?”:
În **dlrlv-ext.conf** avem _A SEC. XVIII^{1}_ ca abreviere. De aceea am mutat `Abbrev::htmlizeAbbreviations` mai sus, să nu trebuiască să reconvertesc în format intern definiția și să compar. Altfel trebuie abrevierea s-o htmlizez și astfel să fac `preg_match`-ul corespunzător.

Reply:
Iar eu mă întrebam de ce nu văd răspunsul tău. :-) (E-mail-ul l-am primit!)

Aha, deci și textul prescurtat poate conține HTML, nu doar cel expandat? Printr-o coincidență fericită am lucrat zilele astea la generalizarea lui `HTML_PATTERNS` din Constant.php. Voi face acum un merge în master. În loc de un simplu șir, înlocuitorul poate fi o funcție (vezi [preg_replace_callback](http://php.net/manual/en/function.preg-replace-callback.php), exemplul 2 pare cel mai clar). Vezi folosirea lui FootnoteHtmlizer prin repo-ul nostru.

Deci aș zice că abrevierile ar trebui prelucrate cu un AbbrevHtmlizer, iar forma scurtă și cea lungă ambele trecute prin `htmlize()`. Avantajul e că poți insera apelul la AbbrevHtmlizer orinde în HTML_PATTERNS (probabil mai spre început, altfel, cum zici și tu, riști ca forma scurtă să sufere modificări).

Fac acum merge-ul.

Reply:
`Str::htmlizeAbbreviations()` nu ajunge spre final, pentru că `$hint` din `title=\"$hint\"` nu e htmlizat, iar lui `self::bestAbbrevMatch` i se pasează `array_keys($abbrevs)`, adică notațiile interne.

Reply:
Am văzut callback-ul, era și în merge-ul anterior.

Reply:
Ok, deci principala întrebare este: unde inserăm `AbbrevHtmlizer` în `HTML_PATTERNS` ca să evităm ciclicitatea?

1. Trebuie **după** conversia `""` -> `„”`, altfel `<span class="abbrev"...` va deveni `<span class=„abbrev”...`.
2. Trebuie **după** `FootnoteHtmlizer`. Dacă o punem înainte, atunci ceva de genul `text{{notă #abr.#}}` va deveni `text {{notă <span class="abbrev">abr.</span>}}`, apoi nota va fi extrasă și HTML-izată a doua oară. Scenariul invers e neplauzibil (abrevieri care să conțină note de subsol).
3. Poate fi **înainte sau după** bold, italic, spațiat. Observația-cheie e că acelea „scuipă” textul de la intrare nemodificat, doar îmbrăcat într-o etichetă.
4. Trebuie să fie **înainte** de conversia `^1` -> `<sup>1</sup>`, căci, cum ziceai, există abrevieri care conțin `^` și pe care nu le mai găsești.

Așadar, propun inserarea imediat după `FootnoteHtmlizer`. Ghilimele vor fi fost deja substituite, deci trebuie ca forma scurtă căutată, dacă are ghilimele, să le aibă pe cele `„”`.

Reply:
Mda, cam așa stăm:
1. creează inconsistență, forma internă a abbrevierilor nu corespunde cu cea a definițiilor privind `„”` vs. `""`
2. da, și mie mi s-a părut cel mai nimerit așa, deși pot fi abrevieri în explicația abrevierilor (ex.: vezi `dlrlv-ext.conf` -> `ms.` = manuscris în multe dintre explicații).
3. depinde cu ce compari, cu reprezentarea internă sau cea html. 
4. depinde cu ce compari, la fel ca în cazul _3._ aldin, cursiv

Reply:
1. Așa e, dar vorbim doar de abrevierile care conțin ghilimele în forma scurtă (pe aceea o cauți).
2. Ei, abrevieri în abrevieri. :-) E ok să le expandăm pe acelea înainte de upload (deci sistemul să presupună că nu există).
3. Voiam să zic că lui `%...%` poți să-i dai HTML și nu-l bușește. De exemplu `%#Expr.#%`, frecvent în DLRLC, poate fi procesat întâi ca `%<span class="abbr">...</span>%' sau întâi ca `<span class="spaced">#Expr.#</span>`. Indiferent în ce ordine procesezi `##` și `%%`, iese bine.
4. Aici contează - dacă faci întâi `^1` -> `<sup>`, atunci la procesarea `##` nu mai găsești.